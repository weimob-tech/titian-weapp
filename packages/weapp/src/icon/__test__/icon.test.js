const { getComponent } = require('../../common/test/getComponent');

describe('test component', () => {
  it('should render success', async () => {
    const comp = getComponent('icon', { name: 'checkbox' });

    const clickCallback = jest.fn();
    comp.addEventListener('click', clickCallback);

    expect(comp).toMatchSnapshot();

    comp.querySelector('.titian-icon').dispatchEvent('tap');
    await Promise.resolve();
    expect(clickCallback).toHaveBeenCalled();
    clickCallback.mockClear();

    comp.setData({ name: 'rotate', color: ['#000', '#ccc'] });
    await Promise.resolve();
    expect(comp).toMatchSnapshot();

    comp.setData({ name: 'rotate', color: 'rgba(0,0,0,1)' });
    await Promise.resolve();
    expect(comp).toMatchSnapshot();
    comp.querySelector('.titian-icon').dispatchEvent('tap');
    await Promise.resolve();
    expect(clickCallback).toHaveBeenCalled();
    clickCallback.mockClear();
  });
});
