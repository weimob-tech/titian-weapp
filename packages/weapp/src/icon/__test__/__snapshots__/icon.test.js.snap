// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <wx-view
    class="titian-icon-base titian-icon titian-icon-checkbox ext-class"
    style=""
    bind:tap="onClick"
  />
</main>
`;

exports[`test component should render success 2`] = `
<main>
  <wx-view
    class="titian-icon-base titian-icon titian-icon-rotate ext-class"
    style="color:#000,#ccc"
    bind:tap="onClick"
  />
</main>
`;

exports[`test component should render success 3`] = `
<main>
  <wx-view
    class="titian-icon-base titian-icon titian-icon-rotate ext-class"
    style="color:rgba(0,0,0,1)"
    bind:tap="onClick"
  />
</main>
`;
